presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationConfig.java

```diff
@@ -263,4 +269,36 @@ public InternalCommunicationConfig setMemoizeDeadNodesEnabled(boolean memoizeDea
         this.memoizeDeadNodesEnabled = memoizeDeadNodesEnabled;
         return this;
     }
+
+    @NotNull
+    public Optional<String> getSharedSecret()
+    {
+        return Optional.ofNullable(sharedSecret);
+    }
+
+    @ConfigSecuritySensitive
+    @Config("internal-communication.shared-secret")
+    public InternalCommunicationConfig setSharedSecret(String sharedSecret)
+    {
+        this.sharedSecret = sharedSecret;
+        return this;
+    }
+
+    public boolean isInternalJwtEnabled()
+    {
+        return internalJwtEnabled;
+    }
+
+    @Config("internal-communication.jwt.enabled")
+    public InternalCommunicationConfig setInternalJwtEnabled(boolean internalJwtEnabled)
+    {
+        this.internalJwtEnabled = internalJwtEnabled;
+        return this;
+    }
+
+    @AssertTrue(message = "When internal JWT(internal-communication.jwt.enabled) authentication is enabled, a shared secret(internal-communication.shared-secret) is required")
+    public boolean isRequiredSharedSecretSet()
+    {
+        return !isInternalJwtEnabled() || getSharedSecret().isPresent();
+    }
```

Comment:
Is there method used anywhere? Also I'm not sure if this is the right place as I couldn't find similar usage.

Reply:
Will remove it.

Reply:
Sorry, this method is annotated with AssertTrue, so it will be checked at server startup. Keeping it.